shared/AppInsightsCommon/src/ThrottleMgr.ts

```diff
@@ -0,0 +1,285 @@
+                }
+            } catch (e) {
+                // eslint-disable-next-line no-empty
+            }
+            return false;
+        }
+
+        // transfer local storage string value to object that identifies start date, current count and preTriggerDate
+        function _getLocalStorageObj(value: string, logger: IDiagnosticLogger, storageName: string) {
+            try {
+                let storageObj = {
+                    date: _getThrottleDate(),
+                    count: 0
+                } as IthrottleLocalStorageObj;
+                if (value) {
+                    let obj = JSON.parse(value);
+                    return {
+                        date: _getThrottleDate(obj.date) || storageObj.date,
+                        count: obj.count || storageObj.count,
+                        preTriggerDate: obj.preTriggerDate? _getThrottleDate(obj.preTriggerDate) : undefined
+                    } as IthrottleLocalStorageObj;
+                } else {
+                    _resetLocalStorage(logger, storageName, storageObj);
+                    return storageObj;
+
+                }
+            } catch(e) {
+                // eslint-disable-next-line no-empty
+            }
+            return null;
+        }
+
+        // if datestr is not defined, current date will be returned
+        function _getThrottleDate(dateStr?: string) {
+            // if new Date() can't be created through the provided dateStr, null will be returned.
+            try {
+                if (dateStr) {
+                    let date = new Date(dateStr);
+                    //make sure it is a valid Date Object
+                    if (!isNaN(date.getDate())) {
+                        return date;
+                    }
+                } else {
+                    return new Date();
+                }
+                
+            } catch (e) {
+                // eslint-disable-next-line no-empty
+            }
+            return null;
+        }
+
+        function _resetLocalStorage(logger: IDiagnosticLogger, storageName: string, obj: IthrottleLocalStorageObj) {
+            try {
+                return utlSetLocalStorage(logger, storageName, strTrim(JSON.stringify(obj)));
+            } catch (e) {
+            //     // eslint-disable-next-line no-empty
+            }
+            return false;
+        }
+
+        function _checkInterval(interval: number, start: number, current: number) {
+            // count from start year
+            return  (current >= start) && (current - start) % interval == 0 ? Math.floor((current - start) / interval) + 1 : -1;
+        }
+        
+        function _sendMessage(msgID: _eInternalMessageId, logger: IDiagnosticLogger, message: string, severity?: eLoggingSeverity) {
+            _throwInternal(logger,
+                severity || eLoggingSeverity.CRITICAL,
+                msgID,
+                message);
+        }
+
+        // NOTE: config.limit.samplingPercentage unit is 1000.
```

Comment:
I'm thinking maybe we should rename this to something like `samplingRate` and in the description we identify this as a `percentage` to the 4 decimal places (I don't think 2 decimal places is enough)

including examples like 1 is the same as specifying 0.0001%